File: 101_200/198_house_robber/Solution.java
class Solution {
    public int rob(int[] nums) {
        int length = nums.length;
        if (length == 0) return 0;
        if (length == 1) return nums[0];
        nums[1] = Math.max(nums[0],nums[1]);
        int tmp;
        for (int i = 2; i < length; ++i) {
            tmp = nums[1];
            nums[1] = Math.max(nums[0] + nums[i], nums[1]);
            nums[0] = tmp;
        }
        return nums[1];
    }
}

File: 101_200/198_house_robber/Solution.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let l = nums.len();
        if l == 0 {
            return 0;
        }
        if l == 1 {
            return nums[0];
        }
        let (mut i, mut a, mut b, mut tmp) = (2usize, nums[0], nums[0].max(nums[1]), 0i32);
        while i < l {
            tmp = b;
            b = b.max(a + nums[i]);
            a = tmp;
            i += 1;
        }
        return b;
    }
}